recipes-app/switchserialmode/files/src/serial/omap_8250.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Siemens AG, 2019-2022
+ *
+ * Authors:
+ *  Gao Nian <nian.gao@siemens.com>
+ *  Chao Zeng <chao.zeng@siemens.com>
+ *
+ * This file is subject to the terms and conditions of the MIT License.  See
+ * COPYING.MIT file in the top-level directory.
+ */
+
```

Comment:
The logic in this file seems common to all tty uart? If so then `tty_uart.c` sounds better to me.

Reply:
for the 8250_omap this is true,we use several extra gpio to control transceiver. I have investigated we can not write a commonly logic to control external transceiver, like mode switch and termination control. 
But the 8250_exar, it can do this.

It can't be guaranteed that all controllers behave in this way, so it's safer to keep the name of the controller

Reply:
Is the omap8250 controlling the transceiver or actually the GPIO pins of the SoC? If later then better to decouple the transceiver toggling out of this 8250 code.

Reply:
for the omap8250 we use the GPIO pins to control the transceiver.  and for the current design, the mode switch for the controller and the transceiver are separated.